packages/ra-core/src/CoreAdmin.tsx

```diff
@@ -15,30 +10,41 @@ import withContext from 'recompose/withContext';
 import createAdminStore from './createAdminStore';
 import TranslationProvider from './i18n/TranslationProvider';
 import CoreAdminRouter from './CoreAdminRouter';
-import { AuthProvider, I18nProvider, DataProvider } from './types';
+import {
+    AuthProvider,
+    I18nProvider,
+    DataProvider,
+    TitleComponent,
+    LoginComponent,
+    LayoutComponent,
+    AdminChildren,
+    CatchAllComponent,
+    CustomRoutes,
+    DashboardComponent,
+} from './types';
 
 export type ChildrenFunction = () => ComponentType[];
 
 interface Props {
-    appLayout: ComponentType;
-    authProvider: AuthProvider;
-    children: ReactNode | ChildrenFunction;
-    catchAll: ComponentType;
-    customSagas: any[];
-    customReducers: object;
-    customRoutes: any[];
-    dashboard: ComponentType;
+    appLayout?: LayoutComponent;
+    authProvider?: AuthProvider;
+    children: AdminChildren;
+    catchAll?: CatchAllComponent;
+    customSagas?: any[];
+    customReducers?: object;
+    customRoutes?: CustomRoutes;
+    dashboard?: DashboardComponent;
     dataProvider: DataProvider;
     history: History;
-    i18nProvider: I18nProvider;
-    initialState: object;
-    loading: ComponentType;
-    locale: string;
-    loginPage: ComponentType | boolean;
-    logoutButton: ComponentType;
-    menu: ComponentType;
-    theme: object;
-    title: ReactNode;
+    i18nProvider?: I18nProvider;
+    initialState?: object;
+    loading?: ComponentType;
+    locale?: string;
+    loginPage?: LoginComponent | boolean;
```

Comment:
the ? here is weird. It's either a component or false, but it can't be undefined. We should make it required and give it a false default value.

Reply:
Making it required means users **have** to provide it. I think it must stay optionnal and we may provide a default `false` value